etna/core/mixins.py

```diff
@@ -89,6 +90,44 @@ def to_dict(self):
         params["_target_"] = BaseMixin._get_target_from_class(self)
         return params
 
+    def set_params(self, **params: dict) -> "BaseMixin":
+        """Return estimator instance with modified parameters.
+
+        The method works on simple estimators as well as on nested objects
+        (such as :class:`~etna.pipeline.Pipeline`). The latter have
+        parameters of the form ``<component>.<parameter>`` so that it's
+        possible to update each component of a nested object.
+
+        Parameters
+        ----------
+        **params: dict
+            Estimator parameters.
+
+        """
+        estimator_out = hydra_slayer.get_from_params(**self.to_dict())
```

Comment:
I still don't think it is a good solution. Imagine that after initialization we define some private fields that depend on create parameters. Later in this code we change public parameters but doesn't change private ones, because we doesn't know about them. We can break the object by doing so.

Reply:
Is there some problem with suggested solution `dict -> patching dict -> creation new object from dict` that we don't know?

Reply:
I don't quite get what `patching dict` means in this context. I've thought that it refers to process of object creation from given dict.
Could you please explain what exactly do you suggest by `pathcing dict`?

Reply:
I discussed this topic with task creator. The idea is the following:
```python
obj = ...
dct = obj.to_dict()
dct.update(**params)
new_obj = Constructor(**dct)
```
If something fails, let it fail and do nothing.

Reply:
> If something fails, let it fail and do nothing.

This contradicts the issue description, namely "check if input dict have unknown parameters we do nothing" line. I think for this statement to hold true we have to check for every parameter in `params` if it is unknown, and then pass only the known parameters to `Constructor`. Should I do this check for existence of parameters?